### DISCUSSION TITLE
Umi4 在约定式路由的情况下，可以支持“wrappers”这个路由属性吗？

### DISCUSSION THREAD

**Discussion by ZN1996** [umijs/umi#10205] [General]
非常喜欢Umi的约定式路由，在Umi3的时候，利用约定式路由搭配`umi-plugin-authorize`这一社区插件，开发体验非常好，既免去了手动维护路由的麻烦，又可以通过配置统一为页面提供对应的权限控制，`umi-plugin-authorize`这个插件原理就是通过`modifyRoutes`这一API，结合配置项去修改匹配路由的`wrappers`属性的。
在升级到Umi4之后，发现`wrappers`这一路由属性在约定式路由的情况下没有生效，不知是这个属性在生成路由阶段被遗漏了？还是`wrappers`属性在约定式路由下不再被支持了？希望大佬们有空可以帮忙解答一下。

**Comment by fz6m** [comment]
约定式路由用嵌套路由就行了。

比如 `/a` 做一个 `index.tsx` ：

```tsx
import { Outlet } from 'umi'

export default function Page() {
  return <Wrapper><Outlet /></Wrapper>
}
```

然后 `/a/some` 写具体的逻辑，这样相当于所有 `/a` 下面都有 wrapper 。

**Reply by ZN1996** [reply]
好的，谢谢你了，我按照你提供的方法再试试看。
